rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/common/CommonRest.java

```diff
@@ -0,0 +1,185 @@
+import org.ow2.proactive_grid_cloud_portal.scheduler.exception.NotConnectedRestException;
+import org.ow2.proactive_grid_cloud_portal.scheduler.exception.RestException;
+import org.ow2.proactive_grid_cloud_portal.scheduler.exception.SchedulerRestException;
+
+
+public class CommonRest implements CommonRestInterface {
+
+    private static final Logger logger = Logger.getLogger(CommonRest.class);
+
+    public static final String YOU_ARE_NOT_CONNECTED_TO_THE_SCHEDULER_YOU_SHOULD_LOG_ON_FIRST = "You are not connected to the scheduler, you should log on first";
+
+    private final SessionStore sessionStore = SharedSessionStore.getInstance();
+
+    private SchedulerStateRest schedulerRest = null;
+
+    private SchedulerRestInterface scheduler() {
+        if (schedulerRest == null) {
+            schedulerRest = new SchedulerStateRest();
+        }
+        return schedulerRest;
+    }
+
+    private String getUserName(String sessionId) throws NotConnectedRestException {
+        Session ss = sessionStore.get(sessionId);
+        return ss.getUserName();
+    }
+
+    @Override
+    public String login(String username, String password) throws LoginException, SchedulerRestException {
+        logger.info("Logging as " + username);
+        return scheduler().login(username, password);
+    }
+
+    @Override
+    public String loginWithCredential(LoginForm multipart) throws KeyException, LoginException, SchedulerRestException {
+        logger.info("Logging using credential file");
+        return scheduler().loginWithCredential(multipart);
+    }
+
+    @Override
+    public void logout(String sessionId) throws RestException {
+        logger.info("logout");
+        scheduler().disconnect(sessionId);
+    }
+
+    @Override
+    public boolean isConnected(String sessionId) {
+        try {
+            getUserName(sessionId);
+            return true;
+        } catch (NotConnectedRestException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public String currentUser(String sessionId) {
+        try {
+            return getUserName(sessionId);
+        } catch (NotConnectedRestException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public UserData currentUserData(String sessionId) {
+        return scheduler().getUserDataFromSessionId(sessionId);
+    }
+
+    @Override
+    public boolean portalAccess(String sessionId, String portal) throws NotConnectedRestException {
+        Scheduler scheduler = checkAccess(sessionId, "EMPTY");
+
+        try {
+            return checkPermission(scheduler.getSubject(),
+                                   new PortalAccessPermission(portal),
+                                   "Acess to portal " + portal + " is disabled");
+        } catch (PermissionException e) {
+            return false;
+        } catch (NotConnectedException e) {
+            throw new NotConnectedRestException("you are not connected to the scheduler, you should log on first");
```

Comment:
@fviale here you should have used `YOU_ARE_NOT_CONNECTED_TO_THE_SCHEDULER_YOU_SHOULD_LOG_ON_FIRST` variable